patricia/patricia.go

```diff
@@ -19,6 +19,29 @@ type (
 	VisitorFunc func(prefix Prefix, item Item) error
 )
 
+const (
+	DefaultMaxPrefixPerNode         = 10
+	DefaultMaxChildrenPerSparseNode = 8
+)
+
+type Option struct {
+	// Max prefix length that is kept in a single trie node.
+	MaxPrefixPerNode int
+	// Max children to keep in a node in the sparse mode.
+	MaxChildrenPerSparseNode int
+}
+
+func NewOption(maxPrefixPerNode int, maxChildrenPerSparseNode int) *Option {
```

Comment:
Not sure we need this function. The usual way is to set only the options that the user wants to modify. I mean, the user doesn't have to use this function, but it's kind of extra...

Anyway, I am wondering if I don't want to implement options in the way as described in http://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis . Let me think about it a bit more...